Dockerfile

```diff
@@ -37,9 +36,30 @@ COPY --from=busybox /bin/mkdir /bin/mkdir
 COPY --from=busybox /bin/chmod /bin/chmod
 COPY --from=busybox /bin/cat /bin/cat
 
-# allow anyone to write to this dir, container may not run as root
-RUN mkdir -p /etc/conjur/ssl && chmod 777 /etc/conjur/ssl
+RUN apk add -u shadow libc6-compat && \
+    # Add limited user
+    groupadd -r secrets-provider \
+             -g 777 && \
+    useradd -c "secrets-provider runner account" \
+            -g secrets-provider \
+            -u 777 \
+            -m \
+            -r \
+            secrets-provider && \
+    # Ensure plugin dir is owned by secrets-provider user
+    mkdir -p /usr/local/lib/secrets-provider /etc/conjur/ssl /run/conjur && \
+    # Use GID of 0 since that is what OpenShift will want to be able to read things
+    chown secrets-provider:0 /usr/local/lib/secrets-provider \
+                           /etc/conjur/ssl \
+                           /run/conjur && \
+    # We need open group permissions in these directories since OpenShift won't
+    # match our UID when we try to write files to them
+    chmod 770 /etc/conjur/ssl \
+              /run/conjur
 
-COPY --from=secrets-provider-builder /opt/secrets-provider-for-k8s/secrets-provider /bin
+USER secrets-provider
+
+COPY --from=secrets-provider-builder /opt/secrets-provider-for-k8s/secrets-provider /usr/local/bin/
+
+CMD [ "/usr/local/bin/secrets-provider"]
```

Comment:
so `CMD` is better than `ENTRYPOINT`? it seemed that the latter is safer from your previous explanation.

Reply:
safer is CMD I would say but I was thinking that we didn't want `secrets-provider` to be overriden because this is the purpose of the build, to run the application. Changing it back because CMD _is_ safer.

Reply:
ENTRYPOINT cannot be overriden and CMD can. What do you means in terms of "Safer"?